internals/overlord/tlsstate/manager.go

```diff
@@ -137,13 +137,23 @@ func (m *TLSManager) ListenConfig() *tls.Config {
 		// not support HTTP2 (it does not yet implement RFC8441). However,
 		// external clients, such as curl, will be able to switch to HTTP2
 		// for non-websocket dependant API calls.
-		NextProtos:     []string{"h2", "http/1.1"},
-		MinVersion:     tls.VersionTLS13,
-		GetCertificate: m.GetCertificate,
+		NextProtos:       []string{"h2", "http/1.1"},
+		MinVersion:       tls.VersionTLS13,
+		GetCertificate:   m.GetCertificate,
+		ClientAuth:       tls.RequestClientCert,
+		VerifyConnection: m.VerifyClientCertificate,
 	}
 	return tlsConf
 }
 
+func (m *TLSManager) VerifyClientCertificate(state tls.ConnectionState) error {
+	certs := len(state.PeerCertificates)
+	if certs != 1 {
+		return fmt.Errorf("expected one client certificate, received %v", certs)
```

Comment:
Nit: prefer %d here as we know the type. Also perhaps `certs` -> `numCerts` for clarity:
```suggestion
		return fmt.Errorf("expected one client certificate, received %d", numCerts)
```

Reply:
Fixed, thank you. Resolving.